src/lib/core/src/identity/player_identity.rs

```diff
@@ -1,6 +1,14 @@
-#[derive(Debug)]
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
```

Comment:
Serializing the player identity for whitelisting is a lot of overkill and would make it harder for server owners to manually edit the whitelist. You could just have the whitelist as a list of usernames and compare a user's name on login, having the entire PlayerIdentity require a match is going to make setting up a whitelist quite hard.